cmd/podman/images/pull.go

```diff
@@ -156,9 +156,21 @@ func imagePull(cmd *cobra.Command, args []string) error {
 		pullOptions.Username = creds.Username
 		pullOptions.Password = creds.Password
 	}
+
+	rawImgName := args[0]
+	// TODO: maybe need update at github.com/containers/image/v5/docker/docker_transport.go:79-81
+	// if use tag and digest, the tag should be ignored
+	if !strings.Contains(rawImgName, "://") {
+		rawImgParts := strings.Split(rawImgName, ":")
+		if len(rawImgParts) == 3 && strings.Contains(rawImgParts[1], "@sha256") {
```

Comment:
There are proper parsing libraries for references. Please don’t add heuristic implementations like these that make hard-to-predict edits on invalid data.